salt/metalk8s/addons/nginx-ingress-control-plane/pre-upgrade.sls

```diff
@@ -0,0 +1,41 @@
+{%- set dest_version = pillar.metalk8s.cluster_version %}
+
+{#- In MetalK8s-2.4.2 we added selector in nginx ingress control plane
+    controller DaemonSet but `selector` are immutable
+    field so, in this case, we cannot replace the object we need to first
+    remove the current one and then deploy the desired one. #}
```

Comment:
@TeddyAndrieux don't we have these selectors set for both control-plane and workload-plane ingress?

If so, we need to remove both objects.. no??

Is there an assumption here that selectors of `workload-plane` ingress are in good shape already before upgrade or downgrade??

Reply:
nginx ingress of `workload-plane` are done in `salt/metalk8s/addons/nginx-ingress/pre-upgrade.sls` here we are in `control-plane` one so we only take care of nginx ingress of control plane

Reply:
yes, true!